compute_endpoint/globus_compute_endpoint/endpoint/config/model.py

```diff
@@ -123,12 +123,12 @@ def _validate(cls, values):
 
         if is_mt:
             msg_engine = "no engine if multi-user"
-            msg_identity = "multi-user requires identity_mapping_config_path"
         else:
             msg_engine = "missing engine"
             msg_identity = "identity_mapping_config_path should not be specified"
         assert is_mt is not bool(values.get("engine")), msg_engine
-        assert is_mt is bool(values.get("identity_mapping_config_path")), msg_identity
+        if not is_mt:
```

Comment:
It looks like we are placing similar checks on privileged users in the EP manager. Does it make sense to consolidate these checks into one place?

Reply:
Hmm, this is a discussion.

From a UX perspective, I don't like Pydantic.  That is, this is scary and doesn't hold the user's hand at all:

```
$ globus-compute-endpoint start example_no_mu_endpoint
Error: 1 validation error for ConfigModel
__root__
  identity_mapping_config_path should not be specified (type=assertion_error)
```

One can figure that out _eventually_, but what the heck is `__root__`?  Why is it an `assertion_error`?  What is a `ConfigModel`?  And after all that, _how do I fix it_?

On the other hand, from a conformity and modeling perspective, it's better to have them in `config/model.py` and Pydantic definitely wins there.

Meanwhile for this specific case, the issue is that it matters in the MU case whether the MEP process is privileged.  If it is, then the `identity_mapping_config_path` item is required.  If it's not privileged, then we don't use it.  For example, it would be weird if one to run the MEP as non-root and be told "nope, not allowed", but then I realize, oh crap, I need to run this as root, then run it via sudo/root and _have to change the configuration_.

So, I've opted to handle this via Pydantic only for the non-MEP case, and then in code for the MEP case.